bond/bond_test.go

```diff
@@ -38,110 +39,108 @@ const (
 			"cniVersion": "%s",
 			"mode": "%s",
 			"failOverMac": 1,
-			"linksInContainer": true,
+			"linksInContainer": %s,
 			"miimon": "100",
-			"mtu": 1400,
+			"mtu": %s,
 			"links": [
 				{"name": "net1"},
 				{"name": "net2"}
 			]
 		}`
 	ActiveBackupMode = "active-backup"
 	BalanceTlbMode   = "balance-tlb"
+	DefaultMTU       = "1400"
 )
 
 var Slaves = []string{Slave1, Slave2}
 
 var _ = Describe("bond plugin", func() {
 	var podNS ns.NetNS
+	var linkAttrs []netlink.LinkAttrs
+	var linksInPodNS bool
+	var hostNS ns.NetNS
 
-	BeforeEach(func() {
+	JustBeforeEach(func() {
 		var err error
 		podNS, err = testutils.NewNS()
 		Expect(err).NotTo(HaveOccurred())
-
-		for _, ifName := range Slaves {
-			err = podNS.Do(func(ns.NetNS) error {
-				defer GinkgoRecover()
-				err = netlink.LinkAdd(&netlink.Dummy{
-					LinkAttrs: netlink.LinkAttrs{
-						Name: ifName,
-					},
-				})
-				Expect(err).NotTo(HaveOccurred())
-				_, err := netlink.LinkByName(ifName)
-				Expect(err).NotTo(HaveOccurred())
-				return nil
-			})
+		if linksInPodNS {
+			addLinksInNS(podNS, linkAttrs)
+		} else {
+			hostNS, err = ns.GetCurrentNS()
+			Expect(err).NotTo(HaveOccurred())
+			addLinksInNS(hostNS, linkAttrs)
 		}
 		Expect(err).NotTo(HaveOccurred())
 	})
 
 	AfterEach(func() {
+		if !linksInPodNS {
+			delLinksInNS(hostNS, linkAttrs)
+			Expect(hostNS.Close()).To(Succeed())
+		}
 		Expect(podNS.Close()).To(Succeed())
 		Expect(testutils.UnmountNS(podNS)).To(Succeed())
 	})
-
-	It("verifies a plugin is added and deleted correctly", func() {
-		args := &skel.CmdArgs{
-			ContainerID: "dummy",
-			Netns:       podNS.Path(),
-			IfName:      IfName,
-			StdinData:   []byte(fmt.Sprintf(Config, "0.3.1", ActiveBackupMode)),
-		}
-
-		err := podNS.Do(func(ns.NetNS) error {
-			defer GinkgoRecover()
+	When("links are in container`s network namespace at initial state (meaning linksInContainer is true)", func() {
```

Comment:
What is the difference between "linksInPodNS" and "linksInContainer" ? 
it seems linksInPodNS is the var used in the test cases right?

Reply:
yes, used "linksInPodNS" as the name of the variable so it matches the test vocabulary (the test included the podNS variable so wanted to match ), but its more important to not be confusing so I changed the variable name  to linksInContainer. think your right and its better to not have 2 names.